src/coreclr/gc/gc.cpp

```diff
@@ -25188,6 +25188,13 @@ void gc_heap::background_promote (Object** ppObject, ScanContext* sc, uint32_t f
     if (o == 0)
         return;
 
+#if defined(FEATURE_CONSERVATIVE_GC) && defined(USE_REGIONS)
```

Comment:
We have similar `is_in_bookkeeping_range` check in `GCHeap::Promote` and `GCHeap::Relocate`. The check is not under `FEATURE_CONSERVATIVE_GC` ifdef there. Is that correct?

Reply:
Suppose the VM always reports either heap or stack addresses when `FEATURE_CONSERVATIVE_GC` is not defined, then we don't need the bookkeeping range check since (not in range) => (on stack) => (no on bookkeeping range) and (in range) => (on heap) => (on bookkeeping range).

Both CoreCLR and NativeAOT has `FEATURE_CONSERVATIVE_GC` defined, are there any build that we don't have the `FEATURE_CONSERVATIVE_GC` defined?

For consistency, I can make sure both sites use the same define conditions.

Reply:
> Suppose the VM always reports either heap or stack addresses when FEATURE_CONSERVATIVE_GC is not defined

What about frozen heap segments?

> Both CoreCLR and NativeAOT has FEATURE_CONSERVATIVE_GC defined, are there any build that we don't have the FEATURE_CONSERVATIVE_GC defined?

We do not currently.

Reply:
Frozen segments are always out of range, so that's fine too.